sdk/keyvault/Azure.Security.KeyVault.Administration/src/RestoreOperation.cs

```diff
@@ -37,20 +37,20 @@ public RestoreOperation(string id, KeyVaultBackupClient client)
             Argument.AssertNotNull(client, nameof(client));
 
             _client = client;
-            _value = new FullRestoreDetailsInternal(string.Empty, string.Empty, null, id, null, null);
+            _value = new RestoreDetailsInternal(string.Empty, string.Empty, null, id, null, null);
         }
 
         /// <summary>
         /// Initializes a new instance of a RestoreOperation.
         /// </summary>
         /// <param name="client">An instance of <see cref="KeyVaultBackupClient" />.</param>
         /// <param name="response">The <see cref="ResponseWithHeaders{T, THeaders}" /> returned from <see cref="KeyVaultBackupClient.StartRestore(Uri, string, string, CancellationToken)"/> or <see cref="KeyVaultBackupClient.StartRestoreAsync(Uri, string, string, CancellationToken)"/>.</param>
-        internal RestoreOperation(KeyVaultBackupClient client, ResponseWithHeaders<FullRestoreDetailsInternal, ServiceFullRestoreOperationHeaders> response)
+        internal RestoreOperation(KeyVaultBackupClient client, ResponseWithHeaders<ServiceFullRestoreOperationHeaders> response)
         {
             Argument.AssertNotNull(client, nameof(client));
             Argument.AssertNotNull(response, nameof(response));
 
-            _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
+            _value = new RestoreDetailsInternal(string.Empty, string.Empty, null, response.Headers.JobId(), null, null);
```

Comment:
See previous comment about allocating sentinel values. Instead, you could also just set/leave `_value` as null and throw if it is, since it's not available.